pypeit/scripts/identify.py

```diff
@@ -86,55 +96,240 @@ def main(args):
+                                    lines_fit_ord = np.array(lines_fit_ord), 
+                                    custom_wav = np.array(custom_wav), 
+                                    custom_wav_ind = np.array(custom_wav_ind) )
+            
+
+        # If we just want the normal one-trace output
         else:
-            waveCalib = None
+            # Load the calibration frame (if it exists and is desired).  Bad-pixel mask
+            # set to any flagged pixel in Arc.
+            wavecal = BuildWaveCalib(msarc, slits, spec, par, lamps, det=args.det,
+                                    msbpm=msarc.select_flag())
+            arccen, arc_maskslit = wavecal.extract_arcs(slitIDs=[int(args.slits)])
 
-        waveCalib.copy_calib_internals(msarc)
+            # Launch the identify window
+            # TODO -- REMOVE THIS HACK
+            try:
+                nonlinear_counts = msarc.detector.nonlinear_counts()
+            except AttributeError:
+                nonlinear_counts = None
+            arcfitter = Identify.initialise(arccen, lamps, slits, slit=int(args.slits), par=par,
+                                            wv_calib_all=wv_calib, wavelim=[args.wmin, args.wmax],
+                                            nonlinear_counts=nonlinear_counts,
+                                            pxtoler=args.pixtol, test=args.test, 
+                                            fwhm=args.fwhm,
+                                            sigdetect=args.sigdetect,
+                                            specname=spec.name,
+                                            y_log=not args.linear,
+                                            rescale_resid=args.rescale_resid)
 
-        # Ask the user if they wish to store the result in PypeIt calibrations
-        arcfitter.store_solution(final_fit, slits.binspec,
-                                 wvcalib=waveCalib,
-                                 rmstol=args.rmstol,
-                                 force_save=args.force_save)
+            # If testing, return now
+            if args.test:
+                return arcfitter
+            final_fit = arcfitter.get_results()
 
+            # Build here to avoid circular import
+            #  Note:  This needs to be duplicated in test_scripts.py
+            # Wavecalib (wanted when dealing with multiple detectors, eg. GMOS)
+            if 'WaveFit' in arcfitter._fitdict.keys():
+                waveCalib = WaveCalib(nslits=1, wv_fits=np.atleast_1d(arcfitter._fitdict['WaveFit']),
+                                    arc_spectra=np.atleast_2d(arcfitter.specdata).T,
+                                    spat_ids=np.atleast_1d(int(arcfitter._spatid)),
+                                    PYP_SPEC=msarc.PYP_SPEC, lamps=','.join(lamps))
+            else:
+                waveCalib = None
 
+            waveCalib.copy_calib_internals(msarc)
+
+        # Ask the user if they wish to store the result in PypeIt calibrations
```

Comment:
indentation error?

Reply:
The  indentation is correct, since there different inputs for the one slit case vs multiple slits.

Reply:
Fixed in my PR